lib/FreeRTOS/freertos-plus-mqtt/freertos_mqtt_agent.c

```diff
@@ -1432,43 +1444,57 @@ MQTTStatus_t MQTTAgent_Init( MQTTContextHandle_t mqttContextHandle,
 
 /*-----------------------------------------------------------*/
 
-MQTTContext_t * MQTTAgent_CommandLoop( void )
+MQTTStatus_t MQTTAgent_CommandLoop( MQTTContextHandle_t * pOutContextHandle,
+                                    MQTTContext_t ** pOutContext )
 {
     Command_t * pCommand;
     MQTTStatus_t operationStatus = MQTTSuccess;
-    MQTTContext_t * ret = NULL;
     CommandType_t currentCommandType = NONE;
 
     /* The command queue should have been created before this task gets created. */
     configASSERT( commandQueue );
 
+    if( ( pOutContextHandle == NULL ) || ( pOutContext == NULL ) )
+    {
+        operationStatus = MQTTBadParameter;
+    }
+
     /* Loop until we receive a terminate command. */
-    for( ; ; )
+    for( ; operationStatus == MQTTSuccess; )
```

Comment:
Would a while loop be more appropriate here?

Reply:
Probably